Sources/PluginLibrary/NamingUtils.swift

```diff
@@ -326,63 +326,61 @@ public enum NamingUtils {
     return prefix + "_"
   }
 
-  /// Remove the proto prefix from the given string.  A proto prefix means
-  /// underscores and letter case are ignored.
-  ///
-  /// - Precodition: The two strings must only be 7bit ascii.
-  ///
-  /// - Returns: nil if nothing can be stripped, otherwise returns the stripping
-  ///            string.
-  static func strip(protoPrefix prefix: String, from: String) -> String? {
-    let prefixChars = prefix.lowercased().unicodeScalars
-    precondition(prefixChars.count == prefix.lengthOfBytes(using: .ascii))
-    var prefixIndex = prefixChars.startIndex
-    let prefixEnd = prefixChars.endIndex
-
-    let fromChars = from.lowercased().unicodeScalars
-    precondition(fromChars.count == from.lengthOfBytes(using: .ascii))
-    var fromIndex = fromChars.startIndex
-    let fromEnd = fromChars.endIndex
-
-    while (prefixIndex != prefixEnd) {
-      if (fromIndex == fromEnd) {
-        // Reached the end of the string while still having prefix to go
-        // nothing to strip.
-        return nil
-      }
+  /// Helper a proto prefix from strings.  A proto prefix means underscores
+  /// and letter case are ignored.
+  struct PrefixStripper {
+    private let prefixChars: String.UnicodeScalarView
+
+    init(prefix: String) {
+      self.prefixChars = prefix.lowercased().replacingOccurrences(of: "_", with: "").unicodeScalars
+    }
+
+    /// Strip the prefix and return the result, or return nil if it can't
+    /// be stripped.
+    func strip(from: String) -> String? {
+      var prefixIndex = prefixChars.startIndex
+      let prefixEnd = prefixChars.endIndex
+
+      let fromChars = from.lowercased().unicodeScalars
+      precondition(fromChars.count == from.lengthOfBytes(using: .ascii))
+      var fromIndex = fromChars.startIndex
+      let fromEnd = fromChars.endIndex
+
+      while (prefixIndex != prefixEnd) {
+        if (fromIndex == fromEnd) {
```

Comment:
Nit: I would personally use `guard`s for many of these early-return checks, but I don't have a strong opinion about it.